flytekit/core/task.py

```diff
@@ -69,58 +69,62 @@ def task(
     **kwargs,
 ) -> Union[Callable, PythonFunctionTask]:
     """
-    This is the core decorator to use for any task type in FlyteKit.
-    Usage: for a simple python task
+    This is the core decorator to use for any task type in flytekit. For a simple python task,
 
-        .. code-block:: python
+    .. code-block:: python
 
-            @task(retries=3)
-            def my_task(x: int, y: typing.Dict[str, str]) -> str:
-                pass
+        @task
+        def my_task(x: int, y: typing.Dict[str, str]) -> str:
+            ...
 
-    Usage: for specific task types
+    For specific task types
 
-        .. code-block:: python
+    .. code-block:: python
 
-            @task(task_config=Spark(), retries=3)
-            def my_task(x: int, y: typing.Dict[str, str]) -> str:
-                pass
+        @task(task_config=Spark(), retries=3)
+        def my_task(x: int, y: typing.Dict[str, str]) -> str:
+            ...
+
+    Please see some cookbook :std:ref:`task examples <cookbook:tasks>` for additional information.
 
     :param _task_function: This argument is implicitly passed and represents the decorated function
     :param task_config: This argument provides configuration for a specific task types.
-                        Please refer to the plugins documentation for the right Object to use
+                        Please refer to the plugins documentation for the right object to use.
     :param cache: Boolean that indicates if caching should be enabled
-    :param cache_version: Version string to be used for the cached value
-    :param retries: for retries=n; n > 0, on failures of this task, the task will be retried at-least n number of times.
+    :param cache_version: Cache version to use. Changes to the task signature will automatically trigger a cache miss,
```

Comment:
Can we add that you *should* bump it for changed task bodies?

Reply:
yes done